web/client/components/misc/FeatureInfoTriggerSelector.jsx

```diff
@@ -25,6 +26,7 @@ class FeatureInfoTriggerSelector extends React.Component {
 
     onChange = (event) => {
         this.props.onSetMapTrigger(event.target.value);
+        this.props.onPurgeMapInfoResults(event.target.value);
```

Comment:
This action takes no argument, please remove it.
Also add a little comment explaining why you are clearing the results when changing trigger event, something like:
// #6870 we clear results because when passing from hover to click the identify panel gets opened (because it has some responses in the current state) while it should not